test/stringio/test_stringio.rb

```diff
@@ -981,4 +981,12 @@ def test_coderange_after_overwrite
   def assert_string(content, encoding, str, mesg = nil)
     assert_equal([content, encoding], [str, str.encoding], mesg)
   end
+
+  if "test".frozen? && !"test".equal?("test") # Ruby 3.4+ chilled strings
+    def test_chilled_string
+      io = StringIO.new("")
+      io << "test"
+      assert_equal("test", s.string)
+    end
+  end
```

Comment:
The file has a frozen string literals comment. This isn't being executed currently.

```suggestion
  if eval('"test".frozen? && !"test".equal?("test")') # Ruby 3.4+ chilled strings
    def test_chilled_string
      io = eval('StringIO.new("")')
      io << "test"
      assert_equal("test", io.string)
    end
  end
```